src/python/pants/bin/daemon_pants_runner.py

```diff
@@ -261,111 +238,58 @@ def nailgunned_stdio(cls, sock, env, handle_stdin=True):
-    # the `pantsd.log`. This ensures that in the event of e.g. a hung but detached pantsd-runner
-    # process that the stacktrace output lands deterministically in a known place vs to a stray
-    # terminal window.
-    # TODO: test the above!
-    ExceptionSink.reset_exiter(self._exiter)
-
-    ExceptionSink.reset_interactive_output_stream(sys.stderr.buffer if PY3 else sys.stderr)
-    ExceptionSink.reset_signal_handler(DaemonSignalHandler())
-
     # Ensure anything referencing sys.argv inherits the Pailgun'd args.
     sys.argv = self._args
 
-    # Set context in the process title.
-    set_process_title('pantsd-runner [{}]'.format(' '.join(self._args)))
-
     # Broadcast our process group ID (in PID form - i.e. negated) to the remote client so
     # they can send signals (e.g. SIGINT) to all processes in the runners process group.
-    NailgunProtocol.send_pid(self._socket, os.getpid())
-    NailgunProtocol.send_pgrp(self._socket, os.getpgrp() * -1)
-
-    # Stop the services that were paused pre-fork.
-    for service in self._services.services:
-      service.terminate()
+    with self._maybe_shutdown_socket.lock:
+      NailgunProtocol.send_pid(self._maybe_shutdown_socket.socket, os.getpid())
+      NailgunProtocol.send_pgrp(self._maybe_shutdown_socket.socket, os.getpgrp() * -1)
 
     # Invoke a Pants run with stdio redirected and a proxied environment.
-    with self.nailgunned_stdio(self._socket, self._env) as finalizer,\
-         hermetic_environment_as(**self._env):
+    with self.nailgunned_stdio(self._maybe_shutdown_socket, self._env) as finalizer, \
+      hermetic_environment_as(**self._env), \
+      encapsulated_global_logger():
       try:
-        # Setup the Exiter's finalizer.
-        self._exiter.set_finalizer(finalizer)
-
         # Clean global state.
         clean_global_runtime_state(reset_subsystem=True)
 
+        # Setup the Exiter's finalizer.
+        self._exiter.set_finalizer(finalizer)
+
         # Otherwise, conduct a normal run.
         runner = LocalPantsRunner.create(
-          self._exiter,
+          NoopExiter(),
           self._args,
           self._env,
           self._target_roots,
           self._graph_helper,
-          self._options_bootstrapper
+          self._options_bootstrapper,
         )
         runner.set_start_time(self._maybe_get_client_start_time_from_env(self._env))
 
-        # Re-raise any deferred exceptions, if present.
-        self._raise_deferred_exc()
-
         runner.run()
       except KeyboardInterrupt:
         self._exiter.exit_and_fail('Interrupted by user.\n')
-      except _GracefulTerminationException as e:
-        ExceptionSink.log_exception(
-          'Encountered graceful termination exception {}; exiting'.format(e))
-        self._exiter.exit(e.exit_code)
-      except Exception:
+      except Exception as e:
+        # LocalPantsRunner.set_start_time resets the global exiter,
```

Comment:
Is this still true? My understanding was that context managers were added to reset the exiters.